go/tasks/pluginmachinery/flytek8s/k8s_resource_adds.go

```diff
@@ -31,6 +31,15 @@ func GetContextEnvVars(ownerCtx context.Context) []v1.EnvVar {
 			},
 		)
 	}
+
+	if nodeID := contextutils.Value(ownerCtx, contextutils.NodeIDKey); nodeID != "" {
```

Comment:
It looks like this ID will be a parent / child hierarchy separated by '/' ([here](https://github.com/flyteorg/flytestdlib/blob/dc81c279d511f271126443b2855843014727e34c/contextutils/context.go#L99-L106)). Do we instead want the nodeID to be identical to that displayed in the UI? We can instead use the `NodeExecutionID.NodeID` and inject [here](https://github.com/flyteorg/flyteplugins/blob/03dc9db2fe948de997734b0d7f2621506a84dd99/go/tasks/pluginmachinery/flytek8s/k8s_resource_adds.go#L73) using something like `id.GetID().NodeExecutionId.NodeId`. Thoughts?

Reply:
Hi @hamersaw 
For our case we need the entire hierarchy, similar to how is is displayed here in the UI:
![image](https://github.com/flyteorg/flyteplugins/assets/26092524/f5d4afd9-9a92-43d4-85f4-12e7557f6240)

(ideally without it being shortened to a specific length)

`id.GetID().NodeExecutionId.NodeId` only gives me the child nodeID, e.g. `dn0`, which might occur multiple times in a workflow. Instead we need the entire hierarchy to uniquely identify a single node in the graph (e.g. `n0/dn0/dn0/dn0/dn0`)

Reply:
If you use something like:
```
currentNodeUniqueID, err := common.GenerateUniqueID(nCtx.ExecutionContext().GetParentInfo(), nodeExecutionID.NodeId)
```
then it should prepend the node ID with the parent info which should give you the `n0-0-dn0` / `n0-0-dn0-0-dn0` being separated by `-` rather than `/`. I think this would be cleaner?